pytensor/xtensor/shape.py

```diff
@@ -506,3 +506,56 @@ def expand_dims(x, dim=None, create_index_for_new_dim=None, axis=None, **dim_kwa
         x = Transpose(dims=tuple(target_dims))(x)
 
     return x
+
+
+class XBroadcast(XOp):
+    """Broadcast multiple XTensorVariables against each other."""
+
+    __props__ = ("exclude",)
+
+    def __init__(self, exclude=None):
+        self.exclude = exclude
+
+    def make_node(self, inputs):
+        # Get the union of all dims in order of first appearance
+        all_dims = []
+        for x in inputs:
+            for d in x.type.dims:
+                if d not in all_dims:
+                    all_dims.append(d)
+
+        # Remove excluded dims
+        if self.exclude:
+            all_dims = [d for d in all_dims if d not in self.exclude]
+
+        # Create a shape map from dim to shape
```

Comment:
Should give priority to non-None static shapes, and check for compatibility. I have a helper for this in the RV PR since it's very commond thing in `make_node`

Reply:
Since it is not yet merged, this is the helper: 
```python
def combine_dims_and_shape(inputs):
    dims_and_shape: dict[str, int | None] = {}
    for inp in inputs:
        for dim, dim_length in zip(inp.type.dims, inp.type.shape):
            if dim not in dims_and_shape:
                dims_and_shape[dim] = dim_length
            elif dim_length is not None:
                # Check for conflicting shapes
                if (dims_and_shape[dim] is not None) and (
                    dims_and_shape[dim] != dim_length
                ):
                    raise ValueError(f"Dimension {dim} has conflicting shapes")
                # Keep the non-None shape
                dims_and_shape[dim] = dim_length
    return dims_and_shape
```